lib/api/contracts/model_contract.rb

```diff
@@ -51,6 +51,27 @@ def self.attribute(*attributes, &block)
       validate :readonly_attributes_unchanged
       validate :run_attribute_validations
 
+      # Performs validation of both, the contract and the underlying model and returns
+      # the merged validation errors for both of them.
+      def validate_contract_and_model
```

Comment:
I dislike the name here as you reveal to much of how you do it. I think you chose the name to avoid conflicting with the validate method that already exists but what is the drawback of overriding it?